libcst/codemod/visitors/_gather_string_annotation_names.py

```diff
@@ -35,14 +35,48 @@ class GatherNamesFromStringAnnotationsVisitor(ContextAwareVisitor):
 
     METADATA_DEPENDENCIES = (QualifiedNameProvider,)
 
-    def __init__(self, context: CodemodContext) -> None:
+    def __init__(
+        self, context: CodemodContext, typing_functions: Optional[Set[str]] = None
+    ) -> None:
         super().__init__(context)
 
+        self._typing_functions: Set[str] = (
+            typing_functions
+            if typing_functions is not None
+            else FUNCS_CONSIDERED_AS_STRING_ANNOTATIONS
+        )
```

Comment:
Why not just make it as default value?
```suggestion
    def __init__(
        self, context: CodemodContext, typing_functions: Optional[Set[str]] = FUNCS_CONSIDERED_AS_STRING_ANNOTATIONS
    ) -> None:
        super().__init__(context)

        self._typing_functions: Set[str] = typing_functions
```

Reply:
Mutable values are not good as default values: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
I think it's a concern when the parameter is mutated in the function which may be a surprise to the users. In our case, it's used as read only parameter which should be fine.

Reply:
If you feel strongly, I can change it; I'm not so sure I'd catch this if a couple months later I make a change that mutates the parameter

Reply:
How about let type checker to make sure it's not mutated? That way it's both simple and safe.
```suggestion
    def __init__(
        self, context: CodemodContext, typing_functions: Collection[str] = FUNCS_CONSIDERED_AS_STRING_ANNOTATIONS
    ) -> None:
        super().__init__(context)
        self._typing_functions: Collection[str] = typing_functions
```

Reply:
That's a great idea, I haven't considered it 👍 Also TIL that `Collection` is immutable

Reply:
Yes, we actually run an automated codemod continuously to refine `Dict` as `Mapping` and `Set` as `Collection` to keep the type annotations more precise.